program_transformers/src/mpl_core_program/v1_asset.rs

```diff
@@ -0,0 +1,558 @@
+        query.sql = format!(
+            "{} WHERE excluded.slot_updated > asset_grouping.slot_updated",
+            query.sql
+        );
+        txn.execute(query)
+            .await
+            .map_err(|db_err| ProgramTransformerError::AssetIndexError(db_err.to_string()))?;
+    }
+
+    //-----------------------
+    // creators table
+    //-----------------------
+
+    let creators = creators
+        .iter()
+        .enumerate()
+        .map(|(i, creator)| asset_creators::ActiveModel {
+            asset_id: ActiveValue::Set(id_vec.clone()),
+            position: ActiveValue::Set(i as i16),
+            creator: ActiveValue::Set(creator.address.to_bytes().to_vec()),
+            share: ActiveValue::Set(creator.percentage as i32),
+            // Note all creators are verified for Core Assets.
+            verified: ActiveValue::Set(true),
+            slot_updated: ActiveValue::Set(Some(slot_i)),
+            seq: ActiveValue::Set(Some(0)),
+            ..Default::default()
+        })
+        .collect::<Vec<_>>();
+
+    if !creators.is_empty() {
+        let mut query = asset_creators::Entity::insert_many(creators)
+            .on_conflict(
+                OnConflict::columns([
+                    asset_creators::Column::AssetId,
+                    asset_creators::Column::Position,
+                ])
+                .update_columns([
+                    asset_creators::Column::Creator,
+                    asset_creators::Column::Share,
+                    asset_creators::Column::Verified,
+                    asset_creators::Column::Seq,
+                    asset_creators::Column::SlotUpdated,
+                ])
+                .to_owned(),
+            )
+            .build(DbBackend::Postgres);
+        query.sql = format!(
+                    "{} WHERE excluded.slot_updated >= asset_creators.slot_updated OR asset_creators.slot_updated is NULL",
+                    query.sql
+                );
+        txn.execute(query)
+            .await
+            .map_err(|db_err| ProgramTransformerError::AssetIndexError(db_err.to_string()))?;
+    }
+
+    // Commit the database transaction.
+    txn.commit().await?;
+
+    // Return early if there is no URI.
+    if uri.is_empty() {
+        warn!(
+            "URI is empty for mint {}. Skipping background task.",
+            bs58::encode(id_vec.clone()).into_string()
+        );
+        return Ok(None);
+    }
+
+    // Otherwise return with info for background downloading.
+    Ok(Some(DownloadMetadataInfo::new(id_vec.clone(), uri)))
+}
+
+async fn find_collection_authority<T: ConnectionTrait + TransactionTrait>(
+    conn: &T,
+    address: Vec<u8>,
+    retry_intervals: &[u64],
+) -> Result<Option<Vec<u8>>, DbErr> {
+    let model: Option<asset_authority::Model> = asset_authority::Entity::find()
+        .filter(asset_authority::Column::AssetId.eq(address.clone()))
+        .one(conn)
+        .await?;
+
+    for interval in retry_intervals {
```

Comment:
Should this loop be at the beginning? Seems like there's a sleep without actually fetching the authority after waking up. This will result in hitting the max retry every time

Reply:
You are right I messed up this logic when copying it over from toke metadata and changing it.  Will fix now.

Reply:
Fixed in https://github.com/metaplex-foundation/digital-asset-rpc-infrastructure/pull/181